apps/web/src/hooks/useRegisterNameCallback.ts

```diff
@@ -62,6 +71,11 @@ export function useRegisterNameCallback(
   );
 
   const [reverseRecord, setReverseRecord] = useState<boolean>(!hasExistingBasename);
+  const [reverseSigPayload, setReverseSigPayload] = useState<{
+    coinTypes: readonly bigint[];
+    signatureExpiry: bigint;
+    signature: `0x${string}`;
+  } | null>(null);
```

Comment:
This useState feels unnecessary since these values are only computed in `registerName` and don't depend on the react lifecycle

Reply:
I switched this to use a ref, which from what I understand helps persist a value across renders though doesn't trigger a re-render if the value changes. Are you thinking we don't even need a ref because the signature will be used immediately upon its creation?

Reply:
> Are you thinking we don't even need a ref because the signature will be used immediately upon its creation?

Yes exactly, we don't need these values to persist across renders since the transaction using the signature is initiated in the same function (`registerName`)

Also, it's generally safer to re-request the signature every time register is clicked because there's currently no cache invalidation implemented. So a flow like this could happen:

1. User clicks register
2. User signs the message
3. User **cancels** the transaction
4. User disconnects and connects a different wallet
5. User clicks register
6. This time there's no signature request and the transaction re-uses the wrong signature from the last wallet